core/src/main/java/com/netflix/conductor/core/execution/tasks/Wait.java

```diff
@@ -29,13 +30,7 @@ public Wait() {
 
     @Override
     public void start(WorkflowModel workflow, TaskModel task, WorkflowExecutor workflowExecutor) {
-        task.setStatus(TaskModel.Status.IN_PROGRESS);
-    }
-
-    @Override
-    public boolean execute(
-            WorkflowModel workflow, TaskModel task, WorkflowExecutor workflowExecutor) {
-        return false;
+        task.setStatus(IN_PROGRESS);
```

Comment:
Is this needed? The `WaitTaskMapper` sets status to IN_PROGRESS when creating the task.

Reply:
Good question :-) I also thought it's not needed. But, when a task is retried, `WorkflowExecutor.retry` creates a TaskModel using `TaskModel.copy()` and sets the status as SCHEDULED. During the decide stage, the `start` method will be called and the status is set to `IN_PROGRESS`.

Currently, there is no clear guideline on where the status should be set and where the associated times (scheduledTime, startTime, endTime) should be set. That's something we need to address.